### PR TITLE
pkg/monitor: add endpoint create and delete monitor notifications

### PR DESCRIPTION
**Pull request by aanm**
It is useful to get monitor events for when an endpoint is added and /
or deleted from Cilium.

Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8728)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

File: pkg/endpointmanager/manager.go [hunk 1]
```diff
@@ -449,15 +450,25 @@ func GetPolicyEndpoints() map[policy.Endpoint]struct{} {
 }
 
 // AddEndpoint takes the prepared endpoint object and starts managing it.
-func AddEndpoint(ep *endpoint.Endpoint, reason string) (err error) {
+func AddEndpoint(owner regeneration.Owner, ep *endpoint.Endpoint, reason string) (err error) {
```
Comment (gandro) [inline pkg/endpointmanager/manager.go:453]: The `owner` is part of `ep` since #8586, no need to pass it separately anymore :)

**Review by gandro — CHANGES_REQUESTED** [review]

File: pkg/endpointmanager/manager.go (hunk 1 above)
Comment (aanm) [reply pkg/endpointmanager/manager.go:453]: @gandro it's a private field of endpoint we need to access it in the `AddEndpoint` function to send the monitor event.

**Comment by aanm** [comment]
test-me-please

File: pkg/endpointmanager/manager.go (hunk 1 above)
Comment (gandro) [reply pkg/endpointmanager/manager.go:453]: Oh, missed the fact that its not accessible. Sorry for the noise then!

**Review by gandro — APPROVED** [review]

**Pull request merged**